compactor2/src/test_util.rs

```diff
@@ -525,6 +527,20 @@ impl TestSetup {
         assert_eq!(file.table_id, self.table.table.id);
         self.table.read_parquet_file(file).await
     }
+
+    /// Set compact version
+    pub fn set_compact_version(&mut self, compact_version: AlgoVersion) {
+        let mut config = Arc::get_mut(&mut self.config).unwrap();
+        config.compact_version = compact_version;
+        self.config = Arc::new(config.clone());
```

Comment:
I don't think you need to both modify the arc and then also clone it 

```suggestion
        let mut config = Arc::get_mut(&mut self.config).unwrap();
        config.compact_version = compact_version;
```